Client/WebAuthN/U2FExtensions.swift

```diff
@@ -979,20 +1186,78 @@ class U2FExtensions: NSObject {
     }
     
     private func cleanupFIDOAuthentication(handle: Int) {
-        u2fActive = false
         guard let index = fidoSignHandles.firstIndex(of: handle) else {
             log.error(U2FErrorMessages.ErrorRegistration)
             return
         }
         fidoSignHandles.remove(at: index)
         fidoSignRequests.removeValue(forKey: handle)
         requestId.removeValue(forKey: handle)
-        YubiKitManager.shared.keySession.cancelCommands()
+        cleanup()
+    }
+    
+    private func executeKeyRequestWith(execution: @escaping () -> Void) {
+        if keyType == .accessory {
+            // Execute the request right away.
+            execution()
+        } else {
+            guard #available(iOS 13.0, *) else {
+                sendJSError()
+                return
+            }
+                        
+            guard let nfcSession = YubiKitManager.shared.nfcSession as? YKFNFCSession else {
+                sendJSError()
+                return
+            }
+            
+            self.nfcActive = true
+            
+            if nfcSession.iso7816SessionState == .open {
+                execution()
+                return
+            }
+            
+            // The ISO7816 session is started only when required since it's blocking the application UI with the NFC system action sheet.
+            YubiKitManager.shared.nfcSession.startIso7816Session()
+            
+            // Execute the request after the key(tag) is connected.
+            nfcSesionStateObservation = nfcSession.observe(\.iso7816SessionState, changeHandler: { [weak self] session, change in
+                if session.iso7816SessionState == .open {
+                    execution()
+                    self?.nfcActive = false
+                    self?.nfcSesionStateObservation = nil // remove the observation
```

Comment:
don't need `return` after this executes?
Or do we want to potentially launch the code below